stage1.js

```diff
@@ -0,0 +1,233 @@
+arr.sort((a, b) => b - a)
+console.log(arr);
+
+//Отфильтровать массив [1,6,7,8,3,4,5,6] по значению >3
+
+const fib = [1, 6, 7, 8, 3, 4, 5, 6]
+const filteredNumbers = fib.filter(num => num > 3)
+console.log(filteredNumbers);
+
+//Написать функцию, которая принимает два параметра - массив и число и выводит элемент
+// массива равный числу
+
+const array = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+function showAlementOfArr(arr, num) {
+  console.log(arr[num]);
+}
+showAlementOfArr(array, 4)
+
+//﻿Реализовать цикл, который будет выводить число а, пока оно не станет меньше 10
+
+let a = 20
+for (let i = 10; i <= a; a--) {
+  console.log(a);
+}
+
+//Реализовать цикл, который выводит в консоль простые числа
+
+nextPrime:
+for (let i = 2; i < 10; i++) {
+  for (let j = 2; j < i; j++) {
+    if (i % j === 0) continue nextPrime
+  }
+  console.log(i);
+}
+
+//Реализовать цикл, который выводит в консоль нечетные числа
+
+const num = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+for (let i = 0; i <= num.length; i++) {
+  const a = num[i];
+  if (a % 2) {
+    console.log(a);
+  }
+}
+
+//Создать 2 объекта: animal и cat, объект animal добавить свойство move,
+//объект cat должен наследовать свойство move
+
+const animal = new Object({
+  cize: 'small',
+  move: function () {
+    console.log('Move!')
+  }
+})
+
+const cat = Object.create(animal)
+cat.name = 'Baby'
+cat.move()
+
+//Дана строка 'ahb acb aeb aeeb adcb axeb'. Напишите регулярное выражение, 
+//которое найдет строки ahb, acb, aeb по шаблону: буква 'a', любой символ, буква 'b'
+
+const str = 'ahb acb aeb aeeb adcb axeb'
+let result = str.match(/a.b/g)
+console.log(result);
+
+//Дана строка '2+3 223 2223'. Напишите регулярку, которая найдет строку 2+3, не захватив остальные
+//1:
+const str1 = '2+3 223 2223'
+let result1 = str1.match(/2\+3/)
+console.log(result1);
+//2:
+console.log('2+3 223 2223'.match(/2\+3/))
+
+//Получить день, месяц и год текущей даты и по отдельности вывести в консоль
+//1:
+const d = new Date();
+
+console.log(d.getFullYear());
+console.log(d.getMonth());
+console.log(d.getDate());
+console.log(d.getDay());
+
+//2:Получить день, месяц и год текущей даты 
+function addLeadingZero(d1) {
+  return (d1 < 10) ? '0' + d1 : d1;
+}
+
+const days = ['Sunday', 'Monday', 'Tuesday', 'Friday', 'Tousday', 'Friday', 'Saturday '];
+
+function getUserTime(t = new Date()) {
+  let Y = t.getFullYear()
+  let M = addLeadingZero(t.getMonth() + 1)
+  let D = addLeadingZero(t.getDate())
+  let d1 = days[t.getDay()]
+
+  console.log(Y, M, D, d1)
+  return `${Y}.${M}.${D} (${d})`
+}
+
+console.log(getUserTime());
+
+//В try catch конструкцию завернуть код: console.log(a), let a = 3.
+// И вывести ошибку – ‘let перед использованием нужно объявить’.
+
+try {
+  console.log(a3);
+  const a3 = 3
+} catch (error) {
+  console.log('Let перед использованием нужно объявить');
+}
+
+// При выполнении 1/0  выводить ошибку 'на ноль делить нельзя'
+
+try {
+  const b = 1
+  const c = 0
+  console.log(b / c);
```

Comment:
In JS we can divide by zero. You need to throw some custom error message if result of the division is `Infinity` and then you need to handle this error.